pallets/deitos/src/lib.rs

```diff
@@ -113,54 +135,180 @@ pub mod pallet {
-            let who = ensure_signed(origin)?;
+        #[pallet::weight(T::WeightInfo::register_ip())]
+        pub fn update_ip_status(
+            origin: OriginFor<T>,
+            provider: AccountIdLookupOf<T>,
+            status: IPStatus,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+            let bounding_amount = IPBoundingAmount::<T>::get()?;
+            let ip = T::Lookup::lookup(provider)?;
+
+            InfrastructureProvider::<T>::try_mutate(
+                &ip,
+                |ip_details| -> Result<_, DispatchError> {
+                    let ip_details = ip_details
+                        .as_mut()
+                        .ok_or(Error::<T>::NonExistentStorageValue)?;
+
+                    ip_details.status = status.clone();
+
+                    Self::deposit_event(Event::IPStatusChanged {
+                        ip: ip.clone(),
+                        status,
+                    });
+
+                    Ok(())
+                },
+            )?;
 
             Ok(())
         }
 
         #[pallet::call_index(2)]
+        #[pallet::weight(T::WeightInfo::update_ip_storage())]
+        pub fn update_ip_storage(
+            origin: OriginFor<T>,
+            price_storage_per_block: BalanceOf<T>,
+            total_storage: Storage,
+        ) -> DispatchResult {
+            // Check that the extrinsic was signed and get the signer.
+            let ip = ensure_signed(origin)?;
+
+            InfrastructureProvider::<T>::try_mutate(
+                &ip,
+                |ip_details| -> Result<_, DispatchError> {
+                    let ip_details = ip_details
+                        .as_mut()
+                        .ok_or(Error::<T>::NonExistentStorageValue)?;
+
+                    // Check if the new total_storage is enough to cover the current reserved_storage
+                    ensure!(
+                        total_storage >= ip_details.reserved_storage,
+                        Error::<T>::InsufficientStorage
+                    );
+
+                    ip_details.price_storage_per_block = price_storage_per_block;
+                    ip_details.total_storage = total_storage;
+                    // Note: This calculation might need to be reviewed or adjusted.
+                    ip_details.reserved_storage = ip_details
+                        .total_storage
+                        .saturating_sub(ip_details.reserved_storage);
+
+                    Self::deposit_event(Event::IPStorageUpdated {
+                        ip: ip.clone(),
+                        price_storage_per_block,
+                        total_storage,
+                    });
+
+                    Ok(())
+                },
+            )?;
+
+            Ok(())
+        }
+
+        #[pallet::call_index(3)]
         #[pallet::weight(T::WeightInfo::shutdown_ip())]
         pub fn shutdown_ip(origin: OriginFor<T>, price_unit: BalanceOf<T>) -> DispatchResult {
```

Comment:
what `price_unit` is needed for?

does `shutdown` mean `unregister`, or is it different?

Reply:
> what price_unit is needed for?

This came from the previous model where IPs were going to be able to define their own price per unit... i need to update this to have one and only price per unit that will be valid for everyone.

> does shutdown mean unregister, or is it different?
i will change it to unregister ... is more clear

Reply:
I think if it removes a IP from state it should be `unregister`.If it just disables an IP then I think it's all fine.